checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -776,9 +788,38 @@ public void visitSwitchExpression17(Tree switchExprTree) {
             checkForNullability(
                     SwitchExpressionUtils.getExpression(switchExprTree), SWITCHING_NULLABLE);
         }
+        if (lintredundantNullComparison) {
+            ExpressionTree expression = SwitchExpressionUtils.getExpression(switchExprTree);
+            List<? extends CaseTree> cases = SwitchExpressionUtils.getCases(switchExprTree);
+            checkSwitchNullRedundant(expression, cases);
+        }
         super.visitSwitchExpression17(switchExprTree);
     }
 
+    /**
+     * Reports an error if the switch expression is @NonNull and the case expression is null
+     * literal.
+     *
+     * @param expression the switch expression
+     * @param cases the case expressions
+     */
+    private void checkSwitchNullRedundant(
+            ExpressionTree expression, List<? extends CaseTree> cases) {
+        AnnotatedTypeMirror switchType = atypeFactory.getAnnotatedType(expression);
+        for (CaseTree caseTree : cases) {
+            List<? extends ExpressionTree> caseExpressions =
+                    TreeUtilsAfterJava11.CaseUtils.getExpressions(caseTree);
+            for (ExpressionTree caseExpression : caseExpressions) {
+                if (caseExpression != null
+                        && caseExpression.getKind() == Tree.Kind.NULL_LITERAL
+                        && switchType.hasEffectiveAnnotation(NONNULL)) {
```

Comment:
You could move this test out of the loop. If the type is not `@NonNull`, the method doesn't need to do anything and can return early.

Reply:
sure!